### PR TITLE
Validate configuration using viper

### LINKED ISSUE #1729: Validate configuration file
Configuration uses `viper`, which is super-easy to use.  Unfortunately it _performs no validation_ of the configuration file.  So e.g. in #1728 the user and I have no way of knowing whether the configuration is valid or not (hint: it contains the invalid configuration key `blockstore.s3.credentials.credentials_file`).

The `viper` API has way of knowing the configuration schema (because everything accessed is optional, and everything _not_ accessed might be accessed later).  Our users probably deserve something that validates.

### PR DESCRIPTION
**Pull request by arielshaqed**
Fixed #1729.

### CODE REVIEW

**Review by nopcoder — APPROVED** [review]

**Comment by arielshaqed** [comment]
Thanks!  Will pull once tests pass.

File: cmd/lakefs/cmd/run.go [hunk 1]
```diff
@@ -55,6 +55,9 @@ var runCmd = &cobra.Command{
 		// validate service names and turn on the right flags
 		dbParams := cfg.GetDatabaseParams()
 
+		logger.Infof("[DEBUG] env %+v", os.Environ())
+		logger.Infof("[DEBUG] dbParams %+v", dbParams)
```
Comment (nopcoder) [inline cmd/lakefs/cmd/run.go]: remove logging

**Review by nopcoder — APPROVED** [review]
LGTM - just one comment about the log

**Comment by nopcoder** [comment]
LGTM!

File: cmd/lakefs/cmd/root.go [hunk 2]
```diff
@@ -35,17 +37,23 @@ func Execute() {
 
 //nolint:gochecknoinits
 func init() {
-	cobra.OnInitialize(initConfig)
+	var initOnce sync.Once
```
[the remark this answers is not in the document]
Comment (arielshaqed) [reply cmd/lakefs/cmd/root.go:40]: Done, thanks!

**Comment by arielshaqed** [comment]
Thanks!

**Pull request merged**